src/devices/onesti.ts

```diff
@@ -1,19 +1,69 @@
-import {Definition} from '../lib/types';
 import * as exposes from '../lib/exposes';
 import fz from '../converters/fromZigbee';
 import tz from '../converters/toZigbee';
+import {Definition, Fz, KeyValue} from 'src/lib/types';
 import * as reporting from '../lib/reporting';
 const e = exposes.presets;
 const ea = exposes.access;
-import * as constants from '../lib/constants';
+
+const fzLocal = {
+    nimly_pro_lock_actions: {
+        cluster: 'closuresDoorLock',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const result: KeyValue = {};
+            const attributes: KeyValue = {};
+            // Handle attribute 257
+            if (msg.data['257'] !== undefined) {
+                const buffer = Buffer.from(msg.data['257']);
+                let pincode = '';
+                for (const byte of buffer) {
+                    pincode += byte.toString(16);
+                }
+                attributes.last_used_pincode = pincode;
+            }
+
+            // Handle attribute 256
+            if (msg.data['256'] !== undefined) {
+                const hex = msg.data['256'].toString(16).padStart(8, '0');
+                const firstOctet = String(hex.substring(0, 2));
+                const lookup: { [key: string]: string } = {
+                    '00': 'MQTT',
```

Comment:
- I proprose to change this to `zigbee`, this library is used by more projects than just Zigbee2MQTT.
- Please make everything lower case here